mtgjson5/providers/tcgplayer.py

```diff
@@ -171,6 +236,62 @@ def generate_tcgplayer_to_mtgjson_map(
     return dump_map
 
 
+def get_tcgplayer_buylist_prices_map(
+    group_id_and_name: Tuple[str, str]
+) -> Dict[str, MtgjsonPricesObject]:
+    """
+    takes a group id and name and finds all buylist data for that group
+    :param group_id_and_name: TCGPlayer Set ID & Name to build
+    :return: returns a map of tcgplayer buylist data to card uuids
+    """
+    LOGGER.info(f"Building {group_id_and_name[1]} TCGPlayer Buylist data")
+    api_response = TCGPlayerProvider().download(
+        f"https://api.tcgplayer.com/pricing/buy/group/{group_id_and_name[0]}"
+    )
+    # breaks if the request is bad
+    if not api_response:
+        return {}
+    response = json.loads(api_response)
+    # breaks if there is no data
+    if not response["results"]:
+        return {}
+    prices_map: Dict[str, MtgjsonPricesObject] = {}
+    # when building the full system, only build this once and take it as an arg/write to a file
+    uuid_map = generate_tcgplayer_to_mtgjson_map(
+        OUTPUT_PATH.joinpath("AllPrintings.json")
+    )
+    # this is set specific, built each time unlike the uuid map
+    sku_map = generate_tcgplayer_sku_map(get_tcgplayer_sku_data(group_id_and_name))
+    for product_buylist_data in response["results"]:
+        # checks if the product id is in the uuid map
+        if not uuid_map.get(str(product_buylist_data["productId"])):
+            continue
+        # if it is in the uuid map we set the key for the pricing object
+        key: str = str(uuid_map.get(str(product_buylist_data["productId"])))
+        # parse each sku to find the near mint skus
+        for sku in product_buylist_data["skus"]:
+            # checks if the sku is NM Nonfoil
+            if sku["skuId"] == sku_map[str(product_buylist_data["productId"])][0]:
+                if sku["prices"]["high"] is not None:
+                    card_price = sku["prices"]["high"]
+                    if key not in prices_map.keys():
+                        prices_map[key] = MtgjsonPricesObject(
+                            "paper", "tcgplayer", TCGPlayerProvider().today_date
+                        )
+
+                    prices_map[key].buy_normal = card_price
+            # checks if the sku is NM Foil
+            elif sku["skuId"] == sku_map[str(product_buylist_data["productId"])][1]:
+                if sku["prices"]["high"] is not None:
+                    card_price = sku["prices"]["high"]
+                    if key not in prices_map.keys():
+                        prices_map[key] = MtgjsonPricesObject(
```

Comment:
Take a closer look at the if/elseif you wrote. If A: (stuff) normal = price elif B: (stuff) foil = price. With this, you can pull (stuff) outside the loops and reduce code complexity. If you're copy/pasting more than 1 line in code, you can (probably) refactor to avoid it :)

Reply:
This is still a problem as you try to make the price object twice. I'd do this as 

for product in response:
   if product id not found continue
   if key not found: make prices object
   for sku in skus:
      if prices.high: if (nonfoil) setNonFoil else setFoil